peerloop/core/dependencies/auth.py

```diff
@@ -0,0 +1,28 @@
+from typing import Annotated
+
+from dependency_injector.wiring import Provide, inject
+from fastapi import Depends
+from fastapi.security import OAuth2PasswordBearer
+
+from peerloop.domain.user.exceptions import UserNotVerifiedError
+from peerloop.domain.user.models import User
+from peerloop.domain.user.service import UserService
+
+oauth_bearer = OAuth2PasswordBearer(tokenUrl="/api/v1/login", scheme_name="JWT")
+
+
+@inject
+async def get_current_user(
+    token: Annotated[str, Depends(oauth_bearer)],
+    user_service: UserService = Depends(Provide["user_container.user_service"]),
+) -> User:
+    user_id = user_service.token_manager.decode_token(token=token)
+    user = await user_service.get_user_by_id(user_id=int(user_id))
+    return user
+
+
+@inject
+async def get_current_verified_user(verified_user: Annotated[User, Depends(get_current_user)]) -> User:
+    if not verified_user.is_verified:
+        raise UserNotVerifiedError("User Not Verified Error: email verification is required.")
+    return verified_user
```

Comment:
What do you think is more appropriate between returning the user model and returning the user id?

I suppose we generally only need `user_id` as a primary key to interact with the database but I'm not sure what's the best practice.

Reply:
I was thinking about it, too. For now, I implemented these functions according to FastAPI document tutorial, which looks good in that it is possible to get basic information about users without additional database access. Most of the time we're going to use this function is when we need basic information about that user!

But I don't know what's the best practice, too...! 😭